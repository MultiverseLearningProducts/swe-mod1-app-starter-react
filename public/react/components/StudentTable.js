import React, { useState, useEffect } from "react";

export const StudentTable = (props) => {

    const [students,setStudents] = useState([])
    const campuses = ["All", "Moon", "Mars", "Saturn"]
    const [filter, setFilter] = useState("All")
    const [allStudents, setAllStudents] = useState([])

    async function fetchStudents(){
		try {
			const response = await fetch('http://localhost:3000/sauces');
			const responseJSON = await response.json();
			setStudents(responseJSON);
            setAllStudents(responseJSON);
		} catch (err) {
			console.log("Oh no an error! ", err)
		}
	}

	useEffect(() => {
		fetchStudents()
	}, []);

    function handleChange (e) {
        const newFilter = e.target.value
        console.log("New filter: ", newFilter)

        // update students based on filter
        if (newFilter == "All"){
            console.log("All students: ", allStudents)
            setStudents(allStudents)
            setFilter(newFilter)
            console.log("Filter: ", filter)
        } else {
            const filteredStudents = allStudents.filter(student => newFilter == student.campus)
            console.log("filtered students: ", filteredStudents)
            setStudents(filteredStudents)
            setFilter(newFilter)
        }
    }

    console.log("students in table: ", students)

    return (
        <>
        <div className="filter-student">
            <label htmlFor="campus-select">Filter Students by Location</label><br/>
            <select id="campus-select" onChange={handleChange}>
                {
                    campuses.map((campus, idx) => {
                        return <option key={idx} value={campus}>{campus}</option>
                    })
                }
            </select>
        </div>
        <div className="table-container">
            <table>
                <thead>
                    <tr>
                        <th>Student ID</th>
                        <th>First Name</th>
                        <th>Last Name</th>
                        <th>Campus</th>
                        <th>Actions</th>
                    </tr>
                </thead>
                <tbody>
                    {   // row for each student
                        students.map((student, idx) => {
                            return (
                                <tr key={idx} value={student} className="student-row">
                                    <td>{student.studentId}</td>
                                    <td>{student.firstName}</td>
                                    <td>{student.lastName}</td>
                                    <td>{student.campus}</td>
                                    <td><a className="table-link" href="">Unenroll</a></td>
                                </tr>
                            )
                        })
                    }
                </tbody>
            </table>
        </div>
        </>
    )


}