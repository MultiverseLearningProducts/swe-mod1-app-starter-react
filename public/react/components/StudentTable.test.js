import React from 'react';
import renderer from 'react-test-renderer';


import { StudentTable } from './StudentTable';
import 'regenerator-runtime/runtime'



// Unit tessting StudenTable

describe("Students Table component", () => {
  test("Contains items passed down as props", () => {
    const studentsList = [{"studentId" : 2, "firstName" : "Ben", "lastName" : "Tate", "campus" : "Mars"}, {"studentId" : 5, "firstName" : "Eric", "lastName" : "Taylor", "campus" : "Mars"}];

    const component = renderer.create(<StudentTable students={studentsList}/>);
    const treeEl = component.toTree()

    expect(treeEl.props.students.length).toBe(2)

  })
  test("renders Student Table component as expected", () => {
    const studentsList = [{"studentId" : 2, "firstName" : "Ben", "lastName" : "Tate", "campus" : "Mars"}, {"studentId" : 5, "firstName" : "Eric", "lastName" : "Taylor", "campus" : "Mars"}];

    const component = renderer.create(<StudentTable students={studentsList}/>);
    const snapshot = component.toJSON();

    console.log("What does our snapshot look like?  ", snapshot)

    expect(snapshot).toMatchSnapshot();

  })
})

