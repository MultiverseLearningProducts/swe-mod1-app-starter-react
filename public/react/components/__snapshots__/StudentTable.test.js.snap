// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students Table component renders Student Table component as expected 1`] = `
Array [
  <div
    className="filter-student"
  >
    <label
      htmlFor="campus-select"
    >
      Filter Students by Location
    </label>
    <br />
    <select
      id="campus-select"
      onChange={[Function]}
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Moon"
      >
        Moon
      </option>
      <option
        value="Mars"
      >
        Mars
      </option>
      <option
        value="Saturn"
      >
        Saturn
      </option>
    </select>
  </div>,
  <div
    className="table-container"
  >
    <table>
      <thead>
        <tr>
          <th>
            Student ID
          </th>
          <th>
            First Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            Campus
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
